src/common/Smi.Common/Messages/MessageHeader.cs

```diff
@@ -73,7 +73,7 @@ public MessageHeader(IDictionary<string, object> encodedHeaders, Encoding enc)
             ProducerProcessID = (int)encodedHeaders["ProducerProcessID"];
             ProducerExecutableName = enc.GetString((byte[])encodedHeaders["ProducerExecutableName"]);
             Parents = GetGuidArrayFromEncodedHeader(encodedHeaders["Parents"], enc);
-            OriginalPublishTimestamp = (long)encodedHeaders["OriginalPublishTimestamp"];
+            OriginalPublishTimestamp = Convert.ToInt64(encodedHeaders["OriginalPublishTimestamp"]); // XXX error casting from Int32 to Int64 using (long)
```

Comment:
Can we clarify this change? The original data type is also a `long` from what I can see.

Reply:
In general Convert is more likely to work than a straight cast so if there was an Exception in it that is resolved by using Convert then its probably fine?

https://stackoverflow.com/questions/15394032/difference-between-casting-and-using-the-convert-to-method

Maybe if the timestamp is low Rabbit serializes it as number without Type?

Reply:
Sure, my question was really "was this a change in response to a bug we found"? I'd prefer not to poke about with the internals of our messaging unless we've actually seen an issue.

Reply:
Resolved offline. This was a change required to handle how Python3 serializes integer types.